src/main/java/org/roaringbitmap/RunContainer.java

```diff
@@ -496,6 +522,38 @@ public Container and(BitmapContainer x) {
         else return answer.toArrayContainer();
     }
 
+    @Override
+    public int andCardinality(BitmapContainer x) {
+        // could be implemented as return toBitmapOrArrayContainer().iand(x);
+        int card = this.getCardinality();
+        int cardinality=0;
+        if (card <=  ArrayContainer.DEFAULT_MAX_SIZE) {
+            // result can only be an array (assuming that we never make a RunContainer)
+            if(card > x.cardinality) card = x.cardinality;
+            cardinality=0;
```

Comment:
This line is not necessary as `cardinality` has already been initialized before and not modified since.